internal/docstore/firedocstore/fs.go

```diff
@@ -168,7 +168,7 @@ func newCollection(client *vkit.Client, collResourceID, nameField string, nameFu
 func (c *collection) Key(doc driver.Document) (interface{}, error) {
 	if c.nameField != "" {
 		name, err := doc.GetField(c.nameField)
-		if err != nil {
+		if err != nil || name == nil {
```

Comment:
I'm not sure about this.

If you wrote
```
map[string]interface{}{"name": nil}
```
then that's probably a bug.

If instead you had a struct and you left the name field blank, then shouldn't we be checking for the zero value? It could be a `string` or `int` field, for example. 

I guess if we check for the zero value, we will have to treat the above situation with a map as a missing key instead of a bug. But that's not too bad.

Reply:
We could also check for zero value besides nil, as we use `interface{}` in the tests.

I was trying to distinguish the two case:
```go
m := map[string]interface{}{"name": nil} // InvalidArgument
s := struct{name interface{}}{} // ok (generate key)
```

Will that be too confusing?

Reply:
I've been trying to ignore the differences between maps and structs, which is why everything happens with `driver.Document`. So I don't think we should distinguish those cases, even though the first is probably a bug. But I'm also worried about
```
struct { Name string }
```
where it won't be nil, it will be `""`

So that's why I think the right thing is to check for the zero value. That will be nil for interface{} and "" for string.

Reply:
Done.